File: 2673.py
# Make Costs of Paths Equal in Binary Tree
# Soln: 139ms, beats 90.48%, 17.86MB, beats 95.24%

class Solution(object):
    def minIncrements(self, n, cost):
        """
        :type n: int
        :type cost: List[int]
        :rtype: int
        """
        ans = 0
        i = ((n + 1) / 2) - 2
        while i >= 0:
            ans += abs(cost[2*i+1] - cost[2*i+2])
            cost[i] += max(cost[2*i+1], cost[2*i+2])
            i -= 1
        return ans

File: 2673.js
// Make Costs of Paths Equal in a Binary Tree
// 5ms beats 100%. 62.83MB beats 100%

/**
 * @param {number} n
 * @param {number[]} cost
 * @return {number}
 */
var minIncrements = function(n, cost) {
    ans = 0;
    for (let i = ( n + 1) / 2 - 2; i >= 0; i--) {
        ans += Math.abs(cost[2*i+1] - cost[2*i+2]);
        cost[i] += Math.max(cost[2*i+1], cost[2*i+2]);
    }
    return ans;
};
